WalletWasabi/Wallets/Wallet.cs

```diff
@@ -457,6 +462,17 @@ private async Task LoadWalletStateAsync(CancellationToken cancel)
 				await ProcessFilterModelAsync(filterModel, cancel).ConfigureAwait(false),
 			new Height(bestKeyManagerHeight.Value + 1),
 			cancel).ConfigureAwait(false);
+
+		// Set BestHeight as the last filter tested.
+		var lastFilter = BitcoinStore.IndexStore.GetLastFilter();
+		if (lastFilter is { })
+		{
+			var maxFilterHeight = new Height(lastFilter.Header.Height);
```

Comment:
You could extract a method.

Reply:
If lastFilter is `null` it means we are in big troubles. The same applies to `LastProcessedFilter` which is not declared as nullable but clearly it (you could take the opportunity to fix that here btw)